rfcs/text/0010-email.md

```diff
@@ -1,28 +1,30 @@
 # 0010: Email
-<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
 
-- Stage: **1 (draft)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
-- Date: **2021-08-16** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+- Stage: **2 (candidate)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
 
 This RFC proposes a new top-level field set to facilitate email use cases, `email.*`. The `email.*` field set adds fields for the sender, recipient, message header fields, and other attributes of an email message typically seen logs produced by mail transfer agent (MTA) and email gateway applications.
 
 ## Fields
 
-<!--
-Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
--->
-
 ### Email specific fields
 
 | field | type | description |
 | --- | --- | --- |
-| `email.from` | keyword | Stores the `from` email address from the RFC5322 `From:` header field. |
+| `email.from.address` | keyword | Stores the `from` email address from the RFC5322 `From:` header field. |
+| `email.from.display_name` | keyword | Stores the display name of the `from` address. |
```

Comment:
Does this not need a nested field because there is only 1 sender?
```
| `email.from` | nested | Nested object with the message sender |
| `email.from.address` | keyword | Stores the `from` email address from the RFC5322 `From:` header field. |
| `email.from.display_name` | keyword | Stores the display name of the `from` address. |
```

Reply:
There's been some back and forth discussion about allowing for multiple `From:` addresses.

Looking back at https://www.rfc-editor.org/rfc/rfc5322#section-3.6.2, IMF _does_ permit multiple addresses in the `From:` field. If multiple addresses appear in the `From:` field, the `Sender:` field is also required but will have a single address.

I see a few possibilities:

1. We keep the `email.from.address` and `email.from.display_name` expecting single values, and then the field description for the `email.from.*` fields provides specific guidance: "if the original event includes multiple `From:` addresses, use the value of the `Sender` field instead."
2. Prefix the field names with `email.sender.*` instead of `email.from`. Like no. 1, the description guides users to use the value of `Sender:` if multiple `From:` values are present.
3. Make `email.from.*` expect a list of objects using `nested` to match `to`, `cc`, and `bcc`. Optionally, discuss if adding `email.sender.*` fields is necessary to discern the `From:` field addresses from the `Sender` field address.

Reply:
> 3. Make email.from.* expect a list of objects using nested to match to, cc, and bcc

I think we should use 3, as it most closely tracks the standard. From the upstream implementation point of view it doesn't add much as it will use the same logic as `to`.

Possibility 1 is incomplete since `more than one` could be > 2, therefore adding a sender field will not cover all cases.

Possibility 2 is confusing to me

> 3. adding email.sender.* fields is necessary

Yes, though these _would_ be single value. Necessary because I have worked on this use case in Legal Search, so I know it exists

Reply:
Opened a PR with these suggestions here: https://github.com/ebeahan/ecs/pull/3

Reply:
> Make email.from.* expect a list of objects using nested to match to, cc, and bcc
> I think we should use 3, as it most closely tracks the standard. From the upstream implementation point of view it doesn't add much as it will use the same logic as to.

I agree with this, I actually had a `suggestion` queued up for this exactly, but wanted to check first.

Reply:
Great, thanks all!

I cherry-picked the changes to include here: https://github.com/elastic/ecs/pull/1593/commits/0133a74574df3c2b306e8744aaf343dcb45f7664